Paul-Sobers/short-response.md

```diff
@@ -0,0 +1,70 @@
+
+   favorite = 'Maya';
+   console.log(`Actually, ${favorite} is my favorite.`);
+   ```
+Answer: When this code is ran `console.log` return the value 'Our favorite Marcy Lab Family member is Juan Pablo!' because `favorite` was a value of 'Juan Pablo' and because it was declared by `const` it cannot be reassigned. So when I try to reassign a value of 'Maya' to `favorite` I recieved an error saying `const variable` was already assigned so when I `console.log` (`Actually,  ${favorite} is my favorite.`); it return the value "Actually, Juan Pablo is my favorite."
+
+
+3. **What does the following code log? Why?**
+   ```javascript
+   let score = 90; 
+   let newScore = score;
+   score = 100; 
+   console.log(score, newScore);
+   ```
+Answer: This code log 100 90 because `score` was assigned the number value 90 and `newScore` was assigned `score` which had the number value 90 also. On line 30 `score` was reassigned a number value of 100 and because `newScore` wasn't reassigned also it kept the number value 90 and `console.log` return 100 90.
+
+4. **Why doesn't the following code log an uppercase string? Change the code so that it does.**
+   ```javascript
+   let greet = 'hello';
+   greet.toUpperCase();
+   let scream = greet;
+   console.log(scream.toUpperCase()); 
+   ```
+Answer: The following code does not log an uppercase string because the `scream` was not attached to the property `.toUpperCase` in the `console.log` on line 40. 
+
+5. **What does the following log? Explain why, ensuring to explain how `someName` and `busy` were declared?**
+   ```javascript
+   let name;
+   let nickname = 'ghost';
+   let someName = name || nickname;
+   let children = null;
+   let homework = 'lots';
+   let busy = children && homework;
+
+   someName ? console.log(`${someName} evaluates to true`) : console.log(`{someName} evaluates to false.`);
+
+   busy ? console.log(`${busy} evaluates to true`) : console.log(`{busy} evaluates to false.``);
+   ```
+Answer: The following code will log ghost evaluates to true and {busy} evaluates to be false. The first log on line 53 evaluates to be true because `someName` was assigned `name` and `nickname` value by the OR logical operator which either value must be true , so on line 53 when `someName ?` was declared it was then asked to choose between a true or false value by the `question mark` ad depending on the trth value it will log one of the following. Since `name ` was undefined(falsey) and `nickname` was assigned a value which made it true the `console.log` logged 'ghost evaluates to true'. The second log {busy} evaluates to be false because on line 55 `busy` was assigned `children` and `homework` by the AND logical operator which both values must be true. So when it was asked (by the `?`) to pick a true or false value it picked `{busy} evaluates to false.` because since the AND logical operator was declared both values (children && homework) must be true in order for the vaule to evaluates to true.
```

Comment:
Yes, Paul! This is logically VERY sound! Great explanation. Language changes below:
> The following code will log `"ghost evaluates to true"` and `"busy evaluates to be false"`. The first log on line 53 evaluates to be true because `someName` was assigned `name` ~~and~~ **OR** `nickname` ~~value by the OR logical operator which either value must be true~~. Since `name ` was undefined (falsey) and `nickname` was assigned a ~~value which made it true~~ **truthy value**,  ~~the `console.log`~~ **the program** logged 'ghost evaluates to true'. The second log**S** `"busy evaluates to ~~be~~ false"` because on line 55 `busy` was assigned `children` and `homework` ~~by~~ using the AND logical operator. **The logical AND shortcircuits on the first falsey value. Since `children` has a falsey value of `null`, we assign busy a value of `null`. Thus, the `else` statement on line 55 will execute.**